forge/src/main/java/org/openjdk/skara/forge/gitlab/GitLabRepository.java

```diff
@@ -751,7 +751,11 @@ public Optional<HostedCommit> commit(Hash hash, boolean includeDiffs) {
                     .onError(r -> Optional.of(JSON.of()))
                     .execute();
             if (!diff.isNull()) {
-                diffs = List.of(toDiff(metadata.parents().get(0), hash, diff));
+                // The diff here is limited by diff patch, diff files count, diff lines.
+                // There is no feasible way to know if this diff is complete.
+                // The diff here is only used to evaluate if a backport PR is clean or not,
+                // assuming the diff is complete will not introduce any side effect.
+                diffs = List.of(toDiff(metadata.parents().get(0), hash, diff, true));
```

Comment:
I thought the diff is only limited by the count of files. But I was wrong.
The diff is limited by 3 factors. So it's hard for us to know if it's limited or not.
https://docs.gitlab.com/administration/diff_limits/#configure-diff-limits

Reply:
I think this is fine. We could perhaps sum files and patch size and compare to the default values (200KB and 1000 files) and at least get a reasonable chance and flagging incomplete when it is.